code/modules/research/machinery/plortmachine.dm

```diff
@@ -0,0 +1,58 @@
+/**********************Plort Redemption Machine**************************/
+//Accepts slime "Cores" as they are definetly NOT called, and turns them into stuff. Probably money? maybe goods? idk im writing this thing before i code it.
+
+/obj/machinery/plortrefinery
+	name = "plort redemption machine"
+	desc = "A machine that accepts slime cores, and sells them to the highest bidder. This generates research, depending on the rarity."
+	icon = 'icons/obj/machines/mining_machines.dmi' // TODO	
+	icon_state = "ore_redemption" // TODO
+	layer = BELOW_OBJ_LAYER
+	density = TRUE
+	speed_process = TRUE
+	var/research_upgrade = 1
```

Comment:
bad variable name, doesn't describe what it does

Reply:
youre a bad variable, but ok
will be fixed